src/high-level/types/complex-double-float.lisp

```diff
@@ -0,0 +1,251 @@
+;;;; complex-double-float.lisp
+;;;;
+;;;; Author: Cole Scott
+
+(in-package #:magicl)
+
+(deftensor tensor/complex-double-float (complex double-float))
+
+(defmatrix matrix/complex-double-float (complex double-float) tensor/complex-double-float)
+
+(defvector vector/complex-double-float (complex double-float) tensor/complex-double-float)
+
+(defcompatible
+    (lambda (tensor)
+      (case (order tensor)
+        (1 '(vector/complex-double-float
+             tensor/complex-double-float))
+        (2 '(matrix/complex-double-float
+             tensor/complex-double-float))
+        (t '(tensor/complex-double-float))))
+  tensor/complex-double-float
+  matrix/complex-double-float
+  vector/complex-double-float)
+
+(defmethod dot ((vector1 vector/complex-double-float) (vector2 vector/complex-double-float))
+  (assert (cl:= (size vector1) (size vector2))
+          () "Vectors must have the same size. The first vector is size ~a and the second vector is size ~a."
+          (size vector1) (size vector2))
+  (loop :for i :below (size vector1)
+        :sum (* (tref vector1 i) (conjugate (tref vector2 i)))))
+
+(defmethod orthonormalize! ((m matrix/complex-double-float))
+  "Applies Gram-Schmidt to the columns of a full rank square matrix to produce a unitary matrix, replacing the elements"
+  (assert-square-matrix m)
+  ;; consider each column
+  (dotimes (j (ncols m))
+    ;; consider each preceding column, which together form an orthonormal set
+    (dotimes (jp j)
+      ;; compute the dot product of the columns...
+      (let ((scalar
+              (loop :for i :below (nrows m)
+                    :sum (* (tref m i j)
+                            (conjugate (tref m i jp))))))
+        ;; ... and do the subtraction.
+        (dotimes (i (nrows m))
+          (setf (tref m i j)
+                (- (tref m i j)
+                   (* scalar
+                      (tref m i jp)))))))
+    ;; now j is orthogonal to the things that came before it. normalize it.
+    (let ((scalar
+            (sqrt
+             (loop :for i :below (nrows m)
+                   :sum (* (abs (tref m i j))
+                           (abs (tref m i j)))))))
+      (dotimes (i (nrows m))
+        (setf (tref m i j)
+              (/ (tref m i j) scalar)))))
+  m)
+
+(defmethod = ((tensor1 tensor/complex-double-float) (tensor2 tensor/complex-double-float) &optional (epsilon +double-comparison-threshold-strict+))
```

Comment:
Note that the default value of `epsilon` in your different implementations of `=` varies. It is zero on some and positive on others.

Reply:
Fixed in 5fecbc2